cpp/include/cudf/utilities/type_dispatcher.hpp

```diff
@@ -592,5 +592,16 @@ CUDF_HOST_DEVICE __forceinline__ constexpr decltype(auto) double_type_dispatcher
                                     std::forward<Ts>(args)...);
 }
 
+/**
+ * @brief Return a name for a given type.
+ *
+ * The returned type names are intended for error messages and are not
+ * guaranteed to be stable.
```

Comment:
Is this comment beneficial to convey intent, or should it be removed?

Reply:
Sort of, the main concern here is that it I'm not sure that this is actually true and is what I was alluding to in [this comment](https://github.com/rapidsai/cudf/issues/11673#issuecomment-1281620304). Right now this works because `type_to_name_impl` is just doing `CUDF_STRINGIFY`, but if that ever changes and the string representation doesn't actually match the name of a type then I think the JIT code paths using this will fail to compile. My (admittedly limited) understanding of jitify2 is that everywhere that is doing `jitify2::reflection::Template(...).instantiate(TYPE1, [TYPE2], ...)` is going to use a string replacement for those arguments to instantiate to instantiate an actual kernel and then compile it, so those types need to map to real type names. Therefore I don't think a function that is suitable for error messages alone, which can be unstable in its output, is suitable for reuse in creating jitify kernels. Correct me if I'm wrong, but I think that these return values need to be real types. I would recommend testing that theory out by changing the behavior for some type and seeing if compilation still succeeds.

Reply:
@vyasr I think I have already tested the premise of your question. The JIT code paths were using `List` and are now using `cudf::list_view` instead. All tests passed.

Reply:
I think you are correct. If the old version was returning arbitrary names, not type names, and it was still working, then my understanding of jitify is incorrect and the current code is fine.

Reply:
OK I was uneasy about approving this without understanding, so I went ahead and traced this back through our jitify stack. This name is indeed not used for actual types.